src/soundbase.cpp

```diff
@@ -247,10 +258,51 @@ void CSoundBase::ParseCommandLineArgument ( const QString& strMIDISetup )
             iCtrlMIDIChannel = slMIDIParams[0].toUInt();
         }
 
+        bool bSimple = true;
+
         // [offset for level]
         if ( slMIDIParams.count() >= 2 )
         {
-            iMIDIOffsetFader = slMIDIParams[1].toUInt();
+            int i = slMIDIParams[1].toUInt ( &bSimple );
+            if ( bSimple )
+                iMIDIOffsetFader = i;
+        }
+
+        if ( bSimple )
+        {
+            for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+            {
+                if ( i + iMIDIOffsetFader > 127 )
+                    break;
+                aMidiCtls[i + iMIDIOffsetFader] = { EMidiCtlType::Fader, i };
+            }
+            return;
+        }
+
+        // We have named controllers
+
+        for ( int i = 1; i < slMIDIParams.count(); i++ )
+        {
+            QString sParm = slMIDIParams[i].trimmed();
+            if ( sParm.isEmpty() )
+                continue;
+
+            int iCtrl = QString ( sMidiCtlChar ).indexOf ( sParm[0] );
+            if ( iCtrl < 0 )
+                continue;
```

Comment:
This is essentially a syntax error in the command line and I think it should be handled as such.

Reply:
There is no code path for handling a syntax error in the controller spec, and the previous implementation ignored everything that was unexpected.  Basically this code is executed significantly after command line parsing and the design prouds itself of passing nothing but a string down in terms of a MIDI spec.  This has been painted as a design goal in previous discussions.

So please propose just where you want the syntax to get checked and what reaction at what point in the code you desire.

Reply:
OK, that's a fair argument.  It's not part of the code I've ever had any interest in before.